### PR TITLE
Add jquery-watch w/ git auto-update via single package.json

### LINKED ISSUE #11756: [Request] Add jquery-watch
**Library name:** jquery-watch
**Library desctiption:** A jQuery plug-in to notify you of CSS, Attribute or Property changes in an element
**Git repository url:**  https://github.com/RickStrahl/jquery-watch
**npm package name or url** (if there is one):  https://www.npmjs.com/package/jquery-watch
**License (List them all if it's multiple):**  MIT
**Official homepage:** http://samples.west-wind.com/jquery-watch/, https://weblog.west-wind.com/posts/2014/Oct/20/A-jquerywatch-Plugin-for-watching-CSS-styles-and-Attributes
**Wanna say something? Leave message here:** 

+++++++++++++++++++++++++++++++++

Very nice and usefull library
Please add it to waiting for popularity tag

### PR DESCRIPTION
**Pull request by sashberd**
close #11756, cc @RickStrahl

Pull request for issue: #11756 
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/RickStrahl/jquery-watch
 * Official website (optional, not the repository): https://weblog.west-wind.com/posts/2014/Oct/20/A-jquerywatch-Plugin-for-watching-CSS-styles-and-Attributes
 * NPM package url (optional): https://www.npmjs.com/package/jquery-watch
 * License and its reference:  MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 10
   - Count of stars:123
   - Count of forks: 24
 * NPM download stats (optional):
   - Downloads in the last day: 3
   - Downloads in the last week: 107
   - Downloads in the last month: 304

# Essential checklist
 * [ ] I'm the author of this library
   * [ X] I would like to add link to the page of this library on CDNJS on website / readme
 * [ X] This lib was not found on cdnjs repo
 * [X ] No already exist / duplicated issue and PR
 * [ ] The lib has notable popularity
   * [ ] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 800 downloads stats per month on npm registry
 * [ X] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ X] Has valid tags for each versions (for git auto-update)
 * [ ] Auto-update setup
 * [ ] Auto-update target/source is valid.
 * [ ] Auto-update filemap is correct.

# Git commit checklist
 * [X ] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [ X] The parent of the commit(s) in the PR is not old than 3 days.
 * [X ] Pull request is sending from a non-master branch with meaningful name.
 * [ X] Separate unrelated changes into different commits.
 * [X ] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [X ] Close corresponding issue in commit message
 * [ X] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by sashberd** [comment]
@sufuf3 
What base path should be, if the files are in the root?

**Comment by PeterDaveHello** [comment]
@sashberd it's possible to have files directly under root.

**Comment by sashberd** [comment]
@PeterDaveHello So the BasePath property should be empty?

**Comment by PeterDaveHello** [comment]
@sashberd in that case files directly under root, yes, basePath should be empty, you can run the auto-updater to validate that (also validate that your pattern is correct). Actually, as I know that you are working in local machine, not the GitHub web gui, you can better commit the latest version in your pull request, the single package.json method is designed for the contributors who has no local working space :)

**Comment by sashberd** [comment]
@PeterDaveHello Any documentation for such work? Because that was Samina and me was doing in our conversation via Gitter

**Comment by PeterDaveHello** [comment]
@sashberd let me know if it's enough:
https://github.com/cdnjs/cdnjs/blob/master/CONTRIBUTING.md#b-pull-request

File: ajax/libs/jquery-watch/package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  },
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/RickStrahl/jquery-watch.git",
+    "fileMap": [
+      {
+	    "basePath": "",
+        "files": [		 
+          "jquery-watch.js",
+          "jquery-watch.min.js"
```
Comment (extend1994) [inline ajax/libs/jquery-watch/package.json]: Let's combine L27 and L28 to `jquery-watch*.+(css|js|map)`. Thank you.

File: ajax/libs/jquery-watch/package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/RickStrahl/jquery-watch"
+  },
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/RickStrahl/jquery-watch.git",
+    "fileMap": [
+      {
+	    "basePath": "",
```
Comment (extend1994) [inline ajax/libs/jquery-watch/package.json]: Some spare spaces here. Please wrap it. You can use `/tools/fixFormat.js` 😄

File: ajax/libs/jquery-watch/package.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{
+  "name": "jquery-watch",
+  "description": "A jQuery plug-in to watch CSS style and attribute changes and get notified when a change occurs",
```
Comment (sufuf3) [inline ajax/libs/jquery-watch/package.json]: Let's modify the description as the same as upstream 
`"This small jQuery plug-in allows you to monitor changes to any DOM element's CSS styles, attributes or properties and fires a callback in response to any change in the monitored styles or attributes."`
We'd like this field as the same with upstream. :)
Thanks.

ref: https://github.com/RickStrahl/jquery-watch/blob/master/package.json#L4

**Pull request merged**